modules/@apostrophecms/attachment/index.js

```diff
@@ -694,11 +694,12 @@ module.exports = {
             // apos.attachment.url with the returned object
             for (let i = ancestors.length - 1; i >= 0; i--) {
               const ancestor = ancestors[i];
-              const fields = ancestor.imagesFields && ancestor.imagesFields[o._id];
-              if (fields) {
+              const ancestorFields = ancestor.attachment && ancestor.attachment._id === value._id && ancestor._fields;
+              if (ancestorFields) {
                 value = _.clone(value);
-                value._crop = _.pick(fields, 'top', 'left', 'width', 'height');
-                value._focalPoint = _.pick(fields, 'x', 'y');
+                o.attachment = value;
```

Comment:
fixes [PRO-2729](https://linear.app/apostrophecms/issue/PRO-2729/annotate-option-of-attachmentall-and-headless-are-incompatible-with) and https://github.com/apostrophecms/apostrophe/pull/3712#issuecomment-1097822579

Reply:
tested [here](https://github.com/apostrophecms/apostrophe/pull/3712/files#diff-b04d3a965201f5f181bf662cc61d52f33e149a0f86b52c470d0a68ded543f240R315)